src/datepicker/bs-datepicker.component.ts

```diff
@@ -70,7 +71,10 @@ export class BsDatepickerDirective implements OnInit, OnDestroy, OnChanges {
       return;
     }
     this._bsValue = value;
-    this.bsValueChange.emit(value);
+
+    if (isValidDate(this._bsValue)) {
```

Comment:
why it's should not notify consumers about date change?

Reply:
Hi, In case of any invalid date entered (like alphabets, or any random string), the entire picker values show invalid date as below,
![image](https://user-images.githubusercontent.com/11587086/72275912-e8da6100-3654-11ea-84ed-a555b87118a0.png)

Hence, to avoid this, when an invalid date is typed, it is not sent further to consumers and stopped here.
#5576 - Please check this for reproduction and expected behavior.

Reply:
sorry can't reproduce